tests/handlers/test_room_summary.py

```diff
@@ -718,6 +717,9 @@ def test_unknown_room_version(self):
         If an room with an unknown room version is encountered it should not cause
         the entire summary to skip.
         """
+        self.hs.get_datastores().main._attempt_to_invalidate_cache(
+            "get_room_version_id", (self.room,)
+        )
```

Comment:
Just for clarity, I would put this after you do the UPDATE below (and maybe add a comment, but I think it'd be more obvious if it came after the update).
The current position of it had me wondering if something was leaking from a previous test or something strange.

I would also write this instead:

`self.hs.get_datastores().main.get_room_version_id.invalidate((self.room,))` — I haven't seen the form you've written before, but the documentation says it will do nothing if the cache doesn't exist. It's probably better to get an error if the cache doesn't exist so we can fix the test :).

Reply:
Done